File: Supercomparator Springboot API/src/main/java/com/tfg/supercomparator/application/service/mercadona/MercadonaService.java
package com.tfg.supercomparator.application.service.mercadona;


import com.tfg.supercomparator.domain.mercadona.MercadonaProduct;
import com.tfg.supercomparator.domain.mercadona.MercadonaProductHistoryItem;
import com.tfg.supercomparator.domain.mercadona.MercadonaProductHistoryItems;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface MercadonaService {
    List<MercadonaProduct> findProduct(String product);

    ResponseEntity<MercadonaProductHistoryItem> saveProductHistory(MercadonaProductHistoryItem mercadonaProductHistory);

    ResponseEntity<MercadonaProductHistoryItems> saveProductsHistory(MercadonaProductHistoryItems mercadonaProductHistory);
}

File: SuperComparator/app/src/main/java/com/tfg/supercomparator/service/MercadonaService.kt
package com.tfg.supercomparator.service

import com.tfg.supercomparator.domain.modules.model.mercadona.MercadonaProduct
import com.tfg.supercomparator.domain.modules.model.mercadona.toProductHistoryItems
import com.tfg.supercomparator.domain.modules.model.product.ProductHistoryItems
import com.tfg.supercomparator.domain.modules.network.MercadonaQuoteService
import com.tfg.supercomparator.domain.modules.network.QuoteRepository
import retrofit2.Response

class MercadonaService : SearchProduct<MercadonaProduct> {
    override suspend fun findProducts(query: String): List<MercadonaProduct> {
        return QuoteRepository
            .getSupercomparatorAPIClient()
            .create(MercadonaQuoteService::class.java)
            .getMercadonaProduct(query)
    }

    override suspend fun saveProductHistory(product: List<MercadonaProduct>): Response<ProductHistoryItems> {
        return QuoteRepository
            .getSupercomparatorAPIClient()
            .create(MercadonaQuoteService::class.java)
            .saveHistory(product.toProductHistoryItems())
    }
}